models/errors_whitebox_test.go

```diff
@@ -0,0 +1,30 @@
+package models
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSimpleError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := simpleError{message: "foo"}
+	assert.Equal(t, "foo", e.Error())
+}
+
+func TestBadParameterError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := BadParameterError{parameter: "foo", value: "bar"}
+	assert.Equal(t, fmt.Sprintf(stBadParameterErrorMsg, e.parameter, e.value), e.Error())
+}
+
+func TestNotFoundError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := NotFoundError{entity: "foo", ID: "bar"}
+	assert.Equal(t, fmt.Sprintf(stNotFoundErrorMsg, e.entity, e.ID), e.Error())
```

Comment:
I think in general what we should test is the intent of our code. The exact text of the error messages is not something the world knows or cares about. Therefore it can change without breaking the program. A meaningful test here would be that we we get back a message of >0 length and not nil, that the code doesn't crash for expected inputs, etc.